### PR TITLE
add const to ctfeexpr.d

### PR DESCRIPTION
**Pull request by WalterBright**

### CODE REVIEW

File: src/dmd/ctfeexpr.d [hunk 1]
```diff
@@ -544,33 +545,41 @@ extern (C++) Expression resolveSlice(Expression e)
...
     {
-        ArrayLiteralExp ale = cast(ArrayLiteralExp)e;
-        return ale.elements ? ale.elements.dim : 0;
-    }
-    if (e.op == TOK.assocArrayLiteral)
-    {
-        AssocArrayLiteralExp ale = cast(AssocArrayLiteralExp)e;
-        return ale.keys.dim;
+        case TOK.vector:
+            return resolveArrayLength((cast(VectorExp)e).e1);
```
Comment (JinShil) [inline src/dmd/ctfeexpr.d]: This looks different.  Is it right?

Comment (JinShil) [reply src/dmd/ctfeexpr.d]: Ah, I see now.  Good!

**Review by JinShil — APPROVED** [review]

**Review by RazvanN7 — APPROVED** [review]

File: src/dmd/ctfeexpr.d [hunk 2]
```diff
@@ -1499,9 +1505,9 @@ extern (C++) Expression findKeyInAA(const ref Loc loc, AssocArrayLiteralExp ae,
      */
     for (size_t i = ae.keys.dim; i;)
     {
-        i--;
+        --i;
```
Comment (UplinkCoder) [inline src/dmd/ctfeexpr.d]: wow!

Comment (WalterBright) [reply src/dmd/ctfeexpr.d]: It is indeed amazing, isn't it?

**Pull request merged**